programs/cryptid_signer/src/instruction/001_propose_transaction.rs

```diff
@@ -24,66 +24,91 @@ impl Instruction for ProposeTransaction {
 
     fn process(
         program_id: Pubkey,
-        mut data: Self::Data,
+        data: Self::Data,
         accounts: &mut Self::Accounts,
     ) -> GeneratorResult<Option<SystemProgram>> {
-        let (key_threshold, settings_sequence) = match &accounts.cryptid_account {
+        let transaction_seeder = TransactionSeeder {
+            cryptid_account: accounts.cryptid_account.info().key,
+            seed: data.account_seed,
+        };
+        let transaction_nonce = transaction_seeder
+            .verify_address_find_nonce(program_id, accounts.transaction_account.info.key)?;
+
+        accounts.cryptid_account.verify_cryptid_account(
+            &program_id,
+            &accounts.did_program.key,
+            &accounts.did.key,
+        )?;
+
+        let (settings_sequence, signer_key, signer_seed_set) = match &accounts.cryptid_account {
             CryptidAccountAddress::OnChain(account) => {
-                (account.key_threshold, account.settings_sequence)
+                let seeder = CryptidSignerSeeder {
+                    cryptid_account: account.info.key,
+                };
+                let signer = seeder.create_address(program_id, account.signer_nonce)?;
+                (
+                    account.settings_sequence,
+                    signer,
+                    PDASeedSet::new(seeder, account.signer_nonce),
+                )
             }
             CryptidAccountAddress::Generative(account) => {
-                CryptidAccountAddress::verify_seeds(
-                    account.key,
-                    program_id,
-                    accounts.did_program.key,
-                    accounts.did.key,
-                )?;
+                let seeder = CryptidSignerSeeder {
+                    cryptid_account: account.key,
+                };
+                let (signer, nonce) = seeder.find_address(program_id);
                 (
-                    CryptidAccount::GENERATIVE_CRYPTID_KEY_THRESHOLD,
                     CryptidAccount::GENERATIVE_CRYPTID_SETTINGS_SEQUENCE,
+                    signer,
+                    PDASeedSet::new(seeder, nonce),
                 )
             }
         };
 
-        accounts
-            .transaction_account
-            .set_funder(accounts.funder.clone());
-        accounts.transaction_account.set_init_size(
-            match NonZeroU64::new(
-                (key_threshold as u64 - data.signers.len() as u64 + data.extra_keyspace as u64)
-                    * size_of::<(Pubkey, UnixTimestamp)>() as u64,
-            ) {
-                None => InitSize::DataSize,
-                Some(non_zero) => InitSize::DataSizePlus(non_zero),
+        *accounts.transaction_account.deref_mut() = TransactionAccount {
+            cryptid_account: accounts.cryptid_account.info().key,
+            accounts: data.accounts,
+            transaction_instructions: data.instructions,
+            signers: accounts
```

Comment:
This is quite a complex functional pipeline. Better to pull it out into a function